mountpoint-s3-fs/src/data_cache/disk_data_cache.rs

```diff
@@ -365,21 +366,18 @@ impl DiskDataCache {
             .recursive(true)
             .create(cache_path_for_key)?;
 
+        let mut temp_file = tempfile::Builder::new()
+            .permissions(fs::Permissions::from_mode(0o600))
+            .tempfile_in(cache_path_for_key)?;
         trace!(
             key = block.header.s3_key,
             offset = block.header.block_offset,
             "writing block at {}",
             path.as_ref().display()
         );
-        let mut file = fs::OpenOptions::new()
-            .write(true)
-            .create(true)
-            .truncate(true)
-            .mode(0o600)
-            .open(path.as_ref())?;
-        file.write_all(CACHE_VERSION.as_bytes())?;
-        let bytes_written = block.write(&mut file)?;
-        Ok(bytes_written)
+        temp_file.write_all(CACHE_VERSION.as_bytes())?;
+        let bytes_written = block.write(&mut temp_file)?;
+        Ok((temp_file, bytes_written))
```

Comment:
The logging here doesn't make sense anymore. We're writing to a temp location.

Maybe we include temp path and final path?